### PR TITLE
Only set CC=mpicc, CXX=mpicxx, F90=mpif90 when necessary

### PR DESCRIPTION
**Pull request by ksagiyam**
This PR is to set CC=mpicc, CXX=mpicxx, and F90=mpif90 only when necessary in the install script, as suggested in https://github.com/firedrakeproject/firedrake/issues/1914.

### CODE REVIEW

File: scripts/firedrake-install [hunk 1]
```diff
@@ -788,7 +806,13 @@ def pip_requirements(package):
...
             return
         else:
-            run_pip_install(["-r", "%s/requirements.txt" % package])
+            reqs = get_requirements("%s/requirements.txt" % package)
+            for req in reqs:
+                if re.split('<|>|=', req)[0] in parallel_packages:
+                    with environment(**parallel_compiler_env):
+                        run_pip_install([req])
+                else:
+                    run_pip_install([req])
```
Comment (dham) [inline scripts/firedrake-install:815]: Please refactor so as to avoid the repeated code.

**Review by dham — CHANGES_REQUESTED** [review]

File: scripts/firedrake-install (hunk 1 above)
Comment (ksagiyam) [reply scripts/firedrake-install]: Thank you. Refactored.

**Comment by wence-** [comment]
I merged the (hopeful) test fix. Please rebase on top of master.

**Comment by ksagiyam** [comment]
Rebased. Thanks!

**Comment by ksagiyam** [comment]
Closing as Jenkins doesn't like this PR. This PR has been replicated at https://github.com/firedrakeproject/firedrake/pull/1926

**Pull request closed without merging**